crates/sui-core/src/authority/authority_store.rs

```diff
@@ -2029,14 +2222,41 @@ impl From<LockDetails> for LockDetailsWrapper {
 
 /// A potential input to a transaction.
 #[derive(Clone, Copy, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
-pub struct InputKey(pub ObjectID, pub Option<SequenceNumber>);
+pub enum InputKey {
+    VersionedObject {
```

Comment:
This might be a good candidate for something to split out and land separately. Though it is probably not important at this point